packages/-ember-data/tests/helpers/qunit-asserts/assert-deprecation.ts

```diff
@@ -0,0 +1,168 @@
+import QUnit from 'qunit';
+import { registerDeprecationHandler } from '@ember/debug';
+import { checkMatcher } from './check-matcher';
+import RSVP from 'rsvp';
+
+let HAS_REGISTERED = false;
+let DEPRECATIONS_FOR_TEST: FoundDeprecation[];
+let HANDLED_DEPRECATIONS_FOR_TEST: FoundDeprecation[];
+
+interface DeprecationConfig {
+  id: string;
+  count?: number;
+  until: string;
+  message?: string | RegExp;
+  url?: string;
+}
+interface FoundDeprecation {
+  message: string;
+  options: {
+    id: string;
+    message?: string;
+    until: string;
+    url?: string;
+  };
+}
+
+interface AssertSomeResult {
+  result: boolean;
+  actual: { id: string; count: number };
+  expected: { id: string; count: number };
+  message: string;
+}
+interface AssertNoneResult {
+  result: boolean;
+  actual: FoundDeprecation[];
+  expected: FoundDeprecation[];
+  message: string;
+}
+
+/**
+ * Returns a qunit assert result object which passes if the given deprecation
+ * `id` was found *exactly* `count` times.
+ *
+ * Fails if not found or found more or less than `count`.
+ * Fails if `until` not specified
+ * Optionally fails if `until` has been passed.
+ */
+function expectDeprecation(config: DeprecationConfig): AssertSomeResult {
```

Comment:
Let's call this method something different, its pretty confusing to have both `QUnit.assert.expectDeprecation` and this inline function with the same name but quite different.

I'm thinking something like `verifyDeprecations`...